packages/core/src/app/ovm/deciders/signed-by-decider.ts

```diff
@@ -0,0 +1,137 @@
+  message: Buffer,
+  signature: Buffer
+) => Promise<boolean>
+
+/**
+ * Decider that determines whether the provided witness is the provided message signed by
+ * the private key associated with the provided public key.
+ */
+export class SignedByDecider extends KeyValueStoreDecider {
+  private static readonly UNIQUE_ID = 'SignedByDecider'
+
+  private readonly signatureVerifier: SignatureVerifier
+
+  constructor(db: DB, signatureVerifier: SignatureVerifier) {
+    super(db)
+
+    this.signatureVerifier = signatureVerifier
+  }
+
+  public async makeDecision(
+    input: SignedByInput,
+    witness: SignedByWitness
+  ): Promise<Decision> {
+    if (!witness) {
+      throw new CannotDecideError(
+        'Cannot decide whether message was signed without witness.'
+      )
+    }
+
+    const signatureMatches: boolean = await this.signatureVerifier(
+      input.publicKey,
+      input.message,
+      witness.signature
+    )
+
+    if (signatureMatches) {
+      await this.storeDecision(
+        input,
+        SignedByDecider.serializeDecision(witness, input, signatureMatches)
+      )
+    }
+
+    return this.constructDecision(
+      witness.signature,
+      input.publicKey,
+      input.message,
+      signatureMatches
+    )
+  }
+
+  protected getUniqueId(): string {
+    return SignedByDecider.UNIQUE_ID
+  }
+
+  protected deserializeDecision(decision: Buffer): Decision {
+    const json: any[] = JSON.parse(decision.toString())
+    return this.constructDecision(
+      Buffer.from(json[0]),
+      Buffer.from(json[1]),
+      Buffer.from(json[2]),
+      json[3]
+    )
+  }
+
+  /**
+   * Builds a Decision from the provided signature, public key, message, and outcome.
+   *
+   * @param signature The signature
+   * @param publicKey The public key used with the signature
+   * @param message The decrypted message
+   * @param outcome the outcome of the Decision
+   * @returns the Decision
+   */
+  private constructDecision(
+    signature: Buffer,
+    publicKey: Buffer,
+    message: Buffer,
+    outcome: boolean
+  ): Decision {
+    return {
+      outcome,
+      justification: [
+        {
+          implication: {
+            decider: this,
+            input: {
+              publicKey,
+              message,
+            },
+          },
+          implicationWitness: {
+            signature,
+          },
+        },
+      ],
+    }
+  }
+
+  /**
+   * Creates the buffer to be stored for a Decision
+   *
+   * @param witness the SignedByWitness
+   * @param input the input that led to the Decision
+   * @param outcome the outcome of the Decision
+   * @returns the Buffer of the serialized data
+   */
+  private static serializeDecision(
+    witness: SignedByWitness,
+    input: SignedByInput,
+    outcome: boolean
+  ): Buffer {
+    return Buffer.from(
+      JSON.stringify([
+        witness.signature.toString(),
+        input.publicKey.toString(),
+        input.message.toString(),
+        outcome,
+      ])
```

Comment:
Could technically use your nice serialization lib `objectToBuffer(...)`

Reply:
Turns out you can't override Buffers' serialization since the Buffer class has a `toJSON(...)` method. That might mark the end of `objectToBuffer` =/